reactApp2/child.jsx

```diff
@@ -1,45 +1,122 @@
+    let value = event.target.value;
+    let index = this.state.array.findIndex((object) => (object.name == value));
+    let item = this.state.array[index];
+    let stateCopy = Object.assign({}, this.state);
+    stateCopy.array[index].isDeleted = true;
+    stateCopy.array[index].Color = 'red';
+    this.setState(stateCopy);
   }
 
-  componentWillMount() {
-      console.log('Component WILL MOUNT!');
-      console.log(this.props.names);
-   }
 
-   componentDidMount() {
-      console.log('Component DID MOUNT!')
-   }
-  componentWillReceiveProps(newProps) {    
-    let names = this.props.names;
+  handleActiveClick(event) {
     let array = this.state.array;
-    
-    if (array.length == 0 || 
-       (!array.find((element) => (names == "" || (element.name == names))))) {
-         let object = {name: names, isDeleted: 0};
-         array.push(object);
-         this.setState({temp : names}); 
+    let newFilterArray = this.state.filterArray;
+    let length = array.length;
+    newFilterArray.splice(0, length);
+    for (let i = 0 ; i < length; i++) {
+      if (!array[i].isDeleted)
+        newFilterArray.push(array[i].name);
     }
+    this.setState({filterArray: newFilterArray });
+    this.setState({state: "Active"});
+  }
+ 
+
+  handleDeletedClick(event) {
+    let array = this.state.array;
+    let newFilterArray = this.state.filterArray;
+    let length = array.length;
+    newFilterArray.splice(0, length);
+    for(let i = 0 ; i < length; i++)
+      if (array[i].isDeleted)
+        newFilterArray.push(array[i].name);
+    this.setState({filterArray: newFilterArray });
+    this.setState({state: "Deleted"});
   }
+ 
 
+  handleAllClick(event) {
+    let array = this.state.array;
+    let newFilterArray = this.state.filterArray;
+    let length = array.length;
+    newFilterArray.splice(0, length);
+    for (let i = 0 ; i < length; i++)
+      newFilterArray.push(array[i].name);
+    this.setState({filterArray: newFilterArray });
+    this.setState({state: "All"});
+  }
+ 
 
   render() {
     return (
       <div>
-         <ul>
-           {this.state.array.map((item,index) => 
-              <li key={index} ><button type = "button" >{item.name}</button>
-              </li>
+        <ul>
+          {this.state.array.map((item,index) => 
+            <li key={index}>
+              <button 
+                type = "button" value = {item.name}
+                onClick = {this.handleClick} 
+                style = {{color:item.Color}}
+                ref = "input">{item.name}
+              </button>             
+            </li>
+          )}      
+        </ul>  
+        <ul>   
+          <button 
+            type="button" 
+            onClick = {this.handleActiveClick}>Active
+          </button>
+          <button 
+            type="button" 
+            onClick = {this.handleDeletedClick}>Deleted
+          </button>
+          <button 
+            type="button" 
+            onClick = {this.handleAllClick}>All
```

Comment:
@VARTIKASINGH1234 I think event variables name should be like following : 
- 'handleActiveClick' to 'displayActiveList'
- 'handleDeletedClick' to 'displayDeletedList'
- 'handleAllClick' to 'displayAllList'